### PR TITLE
fix: the calldata for the upgrades, small fixes for the migration scrips

### PR DESCRIPTION
**Pull request by kelemeno**
## What ❔

-- calldata for upgrades, 
-- scripts for the migration/upgrade
-- bumping contracts to latest release-v23

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.
- [ ] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

**Comment by kelemeno** [comment]
> @kelemeno what's the status of this PR?

I'm using it to prepare the upgrades, txs deploying contracts etc. Does not have features. Once we updated mainnet I will clean it up ( delete the parts that will only be needed for this migration) and merge the necessary components, such as the upgrade txs data

File: etc/upgrades/1711451944-hyperchain-upgrade/localhost/operations.json
Comment (StanislavBreadless) [inline etc/upgrades/1711451944-hyperchain-upgrade/localhost/operations.json:1]: please delete this file

File: infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts [hunk 1]
```diff
@@ -248,5 +287,9 @@ command
             // this is the batch number that the last deposit is processed in ( this is tied to a tx, so commit vs executed is not relevant)
             // we will print the priority tx queue id as part of phase 2 script for local testing, and we can use that to find the batch number
             // for the mainnet upgrade we will have to manually check the priority queue at the given block, since governance is signing the txs
+        } else if (options.addValidators) {
+            await hyperchainUpgradeValidators();
+        } else if (options.whatever) {
```
Comment (StanislavBreadless) [inline infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts:292]: please remove/rename

File: infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts [hunk 2]
```diff
@@ -128,54 +165,54 @@ async function deploySharedBridgeL2Implementation() {
...
+    // env.modify(
+    //     'CONTRACTS_BASE_TOKEN_ADDR',
+    //     '0x0000000000000000000000000000000000000001',
+    //     `etc/env/l2-inits/${process.env.ZKSYNC_ENV}.init.env`
+    // );
+
+    // await deploySharedBridgeL2Implementation();
+
+    // process.chdir(`${process.env.ZKSYNC_HOME}`);
+    // await spawn('zk config compile zksync_local');
```
Comment (StanislavBreadless) [inline infrastructure/protocol-upgrade/src/hyperchain-upgrade.ts:194]: please remove/uncomment this code. alternatively provide a link to the linear task to resolve it

**Review by StanislavBreadless — APPROVED** [review]

**Review by StanislavBreadless — APPROVED** [review]

**Review by mm-zk — APPROVED** [review]

**Pull request merged**